Plugins/Time/NWScript/nwnx_time.nss

```diff
@@ -24,35 +28,33 @@ struct NWNX_Time_HighResTimestamp
     int microseconds; ///< Microseconds
 };
 
+/// @deprecated Use NWNX_Util_GetHighResTimeStamp(). This will be removed in future NWNX releases.
 /// @return Returns the number of microseconds since midnight on January 1, 1970.
 struct NWNX_Time_HighResTimestamp NWNX_Time_GetHighResTimeStamp();
 
 /// @}
 
 string NWNX_Time_GetSystemDate()
 {
-    string sFunc = "GetSystemDate";
-    NWNX_CallFunction(NWNX_Time, sFunc);
-    return NWNX_GetReturnValueString(NWNX_Time, sFunc);
+    WriteTimestampedLogEntry("WARNING:  NWNX_Time is deprecated.  You should migrate to SQLite based functions (see inc_sqlite_time).");
+    return SQLite_GetSystemDate();
 }
 
 string NWNX_Time_GetSystemTime()
 {
-    string sFunc = "GetSystemTime";
-    NWNX_CallFunction(NWNX_Time, sFunc);
-    return NWNX_GetReturnValueString(NWNX_Time, sFunc);
+    WriteTimestampedLogEntry("WARNING:  NWNX_Time is deprecated.  You should migrate to SQLite based functions (see inc_sqlite_time).");
+    return SQLite_GetSystemTime();
 }
 
 int NWNX_Time_GetTimeStamp()
 {
-    string sFunc = "GetTimeStamp";
-
-    NWNX_CallFunction(NWNX_Time, sFunc);
-    return NWNX_GetReturnValueInt(NWNX_Time, sFunc);
+    WriteTimestampedLogEntry("WARNING:  NWNX_Time is deprecated.  You should migrate to SQLite based functions (see inc_sqlite_time).");
+    return SQLite_GetTimeStamp();
 }
 
 struct NWNX_Time_HighResTimestamp NWNX_Time_GetHighResTimeStamp()
 {
+    WriteTimestampedLogEntry("WARNING:  NWNX_Time is deprecated.  NWNX_Time_GetHighResTimeStamp is moving to NWNX_Utility.");
     struct NWNX_Time_HighResTimestamp t;
     string sFunc = "GetHighResTimeStamp";
 
```

Comment:
Don't we want to move the call to NWNX_Util_GetHighResTimestamp() here?

Reply:
I noticed I wrote NWNX_Utility instead of NWNX_Util. I'll correct that.

Changing to calling the function in NWNX_Util there would mean it would stop working for someone who does not have the util plugin enabled I think?

Reply:
Yeah, but only if they update the nss. If they're already updating the nss, they can just enable Util.

Also, I have a hard time imagining someone using this time function but not having Util enabled.

Reply:
I made the change (and rebased on current master and cleaned up commits a bit).